pyanaconda/modules/payload/base/source_base_interface.py

```diff
@@ -0,0 +1,51 @@
+#
+# Base object of all payload sources.
+#
+# Copyright (C) 2019 Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+from abc import ABCMeta
+
+from pyanaconda.dbus.interface import dbus_interface
+from pyanaconda.dbus.typing import *  # pylint: disable=wildcard-import
+from pyanaconda.modules.common.base.base_template import ModuleInterfaceTemplate
+from pyanaconda.modules.common.constants.interfaces import PAYLOAD_SOURCE
+
+
+@dbus_interface(PAYLOAD_SOURCE.interface_name)
+class PayloadSourceBaseInterface(ModuleInterfaceTemplate, metaclass=ABCMeta):
+    """Base class for all the payload source module interfaces.
+
+    This object contains all API shared for all the sources. Everything in this object have
```

Comment:
"... contains API shared by all the sources... " ?
"... has to be implemented... " ?

Reply:
Fixed.